include/boost/geometry/algorithms/detail/is_valid/pointlike.hpp

```diff
@@ -42,14 +45,26 @@ struct is_valid<Point, point_tag>
 // (have identical coordinate values in X and Y)
 //
 // Reference: OGC 06-103r4 (6.1.5)
-template <typename MultiPoint>
-struct is_valid<MultiPoint, multi_point_tag>
+template <typename MultiPoint, bool AllowEmptyMultiGeometries>
+struct is_valid<MultiPoint, multi_point_tag, AllowEmptyMultiGeometries>
 {
-    static inline bool apply(MultiPoint const& /*multipoint*/)
+    template <typename VisitPolicy>
+    static inline bool apply(MultiPoint const& multipoint,
+                             VisitPolicy& visitor)
     {
-        // we allow empty multi-geometries, so an empty multipoint
-        // is considered valid
-        return true;
+        if (AllowEmptyMultiGeometries || boost::size(multipoint) > 0)
+        {
+            // we allow empty multi-geometries, so an empty multipoint
+            // is considered valid
+            visitor.template apply<no_failure>();
+            return true;
+        }
+        else
+        {
+            // we do not allow an empty multipoint
+            visitor.template apply<failure_few_points>();
+            return false;
```

Comment:
OK, but it can be an advantage too, to have a specific visitor that copies normal behaviour but for one specific situation it reverses the result...
It is for me not necessary to avoid the if, I was just considering letting the visitor return the result indeed. If you considered this and still like it, OK for me.